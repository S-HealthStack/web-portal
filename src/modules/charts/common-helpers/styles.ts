import styled, { css } from 'styled-components';

import { colors, typography, px, animation } from 'src/styles';
import { LINE_CLASS_NAME } from '../common-components/Line';
import { AREA_CLASS_NAME } from '../common-components/Area';
import { DOT_CLASS_NAME } from '../common-components/Dot';

interface ContainerProps {
  width: number;
  height: number;
}

export const DOTS_CONTAINER_CLASS_NAME = 'dots';
export const X_AXIS_CLASS_NAME = 'xAxis';
export const Y_AXIS_CLASS_NAME = 'yAxis';
export const ZOOM_CLASS_NAME = 'zoom';

export const Container = styled.div.attrs<ContainerProps>(({ width, height }) => ({
  style: {
    width: px(width),
    height: px(height),
    position: 'relative',
  },
}))<ContainerProps>``;

export const TooltipContent = styled.div`
  display: flex;
  align-items: center;
`;

export const TooltipNameBlock = styled.div`
  display: flex;
  margin-right: ${px(8)};

  p {
    ${typography.labelSemibold};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipDataBlock = styled.div`
  display: flex;
  flex-direction: column;

  p {
    ${typography.labelRegular};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipColorPoint = styled.div<{ color: string }>`
  width: ${px(12)};
  height: ${px(12)};
  background-color: ${({ color }) => color};
  border-radius: 50%;
  margin-right: ${px(8)};
`;

export const brushOverlayStyles = css`
  fill: ${colors.primary};
  fill-opacity: 0.1;
  pointer-events: none;
`;

export const brushSelectionStyles = css`
  fill: ${colors.primary};
  fill-opacity: 1;
`;

export const brushHandleCommonStyles = css`
  stroke-width: ${px(1)};
  stroke: ${colors.primary};
  stroke-opacity: 0.3;
  fill: ${colors.onPrimary};
`;

export const StyledSvg = styled.svg<{
  focusWidth: number;
  $contextVisible: boolean;
}>`
  .${X_AXIS_CLASS_NAME} {
    .tick line {
      opacity: 0;
    }

    .tick {
      :first-child text {
        text-anchor: start;
      }
      :last-child text {
        text-anchor: end;
      }
    }
  }

  .${Y_AXIS_CLASS_NAME} {
    .tick line {
      stroke: ${colors.updBackgoundLight};
    }
  }

  .context {
    opacity: ${({ $contextVisible }) => ($contextVisible ? 1 : 0)};
    transition: opacity 0.3s ${animation.defaultTiming};
    .brushX {
      .overlay {
        ${brushOverlayStyles};
        height: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        height: ${px(22)};
        fill: url(#horizontalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(22)};
        width: ${px(6)};
      }
    }
    .brushY {
      .overlay {
        ${brushOverlayStyles};
        width: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        width: ${px(22)};
        fill: url(#verticalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(6)};
        width: ${px(22)};
      }
    }
  }

  .focus {
    cursor: move;
  }

  .${LINE_CLASS_NAME} {
    clip-path: url(#clip);
    transition: stroke-opacity 0.3s ${animation.defaultTiming};
  }

  .${AREA_CLASS_NAME} {
    clip-path: url(#clip);
    transition: fill-opacity 0.3s ${animation.defaultTiming};
  }

  .${DOTS_CONTAINER_CLASS_NAME} {
    clip-path: url(#clip);
    .${DOT_CLASS_NAME} {
      transition: fill-opacity 0.3s ${animation.defaultTiming};
    }
  }

  .zoom {
    fill: none;
    pointer-events: all;
  }
`;
