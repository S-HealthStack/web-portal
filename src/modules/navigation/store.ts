import { createSelector } from '@reduxjs/toolkit';
import { matchPath } from 'react-router-dom';
import { createBrowserHistory, createMemoryHistory, Location } from 'history';

import { RootState } from 'src/modules/store/store';

export enum Path {
  Root = '/',

  SignIn = '/signin',
  AccountActivation = '/account/activation',
  AccountCreate = '/account/create',
  AccountConfirm = '/account/confirm',
  AccountVerification = '/email-verification',
  ForgotPassword = '/account/forgot',
  ForgotPasswordConfirm = '/account/forgot/confirm',
  ResetPassword = '/account/forgot/reset',
  ResetPasswordComplete = '/account/forgot/complete',

  CreateStudy = '/create-study',

  Overview = '/overview',
  OverviewSubject = '/overview/participants/:subjectId',
  TrialManagement = '/trial-management',
  TrialManagementSubject = '/trial-management/participants/:subjectId',
  TrialManagementEditSurvey = '/trial-management/surveys/:surveyId/edit',
  TrialManagementSurveyResults = '/trial-management/surveys/:surveyId/results',
  UserAnalytics = '/user-analytics',
  DataCollection = '/data-collection',
  DataCollectionSubject = '/data-collection/participants/:subjectId',
  StudySettings = '/study-settings',
}

export const makeHistory = () =>
  process.env.NODE_ENV !== 'test'
    ? createBrowserHistory({ basename: process.env.PUBLIC_PATH })
    : createMemoryHistory();

export const history = makeHistory();

const sectionPaths = [
  Path.Overview,
  Path.OverviewSubject,
  Path.TrialManagement,
  Path.UserAnalytics,
  Path.DataCollection,
  Path.StudySettings,
] as const;

export const sectionPathSelector = createSelector(
  (state: RootState) => state.router,
  (r) => sectionPaths.find((s) => matchPath((r.location as unknown as Location).pathname, s))
);
