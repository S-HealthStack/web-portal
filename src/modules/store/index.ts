export type { RootState, AppDispatch, AppThunk } from './store';
export { useAppDispatch, useAppSelector } from './hooks';
export * from './tools';
