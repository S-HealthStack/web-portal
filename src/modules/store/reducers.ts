import { combineReducers } from '@reduxjs/toolkit';
import { connectRouter } from 'connected-react-router';
import type { Reducer } from 'redux';

import authReducer from 'src/modules/auth/auth.slice';
import studySettingsReducer from 'src/modules/study-settings/studySettings.slice';
import snackbarReducer from 'src/modules/snackbar/snackbar.slice';
import overviewReducers from 'src/modules/overview/overview.slice';
import studiesReducer from 'src/modules/studies/studies.slice';
import studyManagementReducer from 'src/modules/study-management/studyManagement.slice';
import dataCollectionReducer from 'src/modules/data-collection/dataCollection.slice';
import sidebarReducer from 'src/modules/main-layout/sidebar/sidebar.slice';
import fileUploadReducer from 'src/modules/file-upload/fileUpload.slice';
import { history } from 'src/modules/navigation/store';
import { DataSliceState } from 'src/modules/store/createDataSlice';
import forgotPasswordReducers from 'src/modules/auth/forgot-password/forgotPassword.slice';
import resetPasswordReducers from 'src/modules/auth/forgot-password/resetPassword.slice';
import objectStorageReducers from 'src/modules/object-storage/objectStorage.slice';

const createReducer = <T, V>(asyncReducers?: { [key: string]: Reducer<DataSliceState<T, V>> }) =>
  combineReducers({
    ...authReducer,
    router: connectRouter(history),
    snackbar: snackbarReducer,
    dataCollection: dataCollectionReducer,
    sidebar: sidebarReducer,
    ...studySettingsReducer,
    ...overviewReducers,
    ...studyManagementReducer,
    ...forgotPasswordReducers,
    ...resetPasswordReducers,
    ...fileUploadReducer,
    ...asyncReducers,
    ...objectStorageReducers,
    ...studiesReducer,
  });

export default createReducer;
