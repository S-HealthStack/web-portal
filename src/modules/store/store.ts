/* eslint-disable import/no-import-module-exports */
import {
  Action,
  ThunkAction,
  configureStore,
  getDefaultMiddleware,
  ThunkDispatch,
} from '@reduxjs/toolkit';
import { routerMiddleware } from 'connected-react-router';

import { history } from 'src/modules/navigation/store';
import rootReducer from 'src/modules/store/reducers';

export { rootReducer };

export const store = configureStore({
  reducer: rootReducer,
  middleware: [...getDefaultMiddleware(), routerMiddleware(history)],
  devTools: process.env.NODE_ENV !== 'production',
});

if (module.hot) {
  // Enable Webpack hot module replacement for reducers
  module.hot.accept('./reducers', async () => {
    // eslint-disable-next-line import/no-useless-path-segments
    const nextRootReducer = (await import('../store/reducers')).default;
    store.replaceReducer(nextRootReducer);
  });
}

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = ThunkDispatch<RootState, unknown, Action>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
