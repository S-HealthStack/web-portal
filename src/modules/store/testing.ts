import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { rootReducer, RootState } from './store';

export const createTestStore = (state: ConfigureStoreOptions<RootState>['preloadedState']) =>
  configureStore({
    reducer: rootReducer(),
    preloadedState: state,
  });
