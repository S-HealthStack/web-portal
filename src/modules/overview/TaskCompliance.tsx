import React, { useEffect, useMemo, useRef, useState } from 'react';
import useScrolling from 'react-use/lib/useScrolling';

import _round from 'lodash/round';
import styled from 'styled-components';

import { TaskType } from 'src/modules/api';
import RankIcon from 'src/assets/icons/rank.svg';
import SortAscIcon from 'src/assets/icons/sort_asc.svg';
import SortDescIcon from 'src/assets/icons/sort_dsc.svg';
import { withCustomScrollBar } from 'src/common/components/CustomScrollbar';
import ProgressBar from 'src/common/components/ProgressBar';
import Card, { Title } from 'src/common/components/Card';
import OverviewCardWrapperWithSkeleton from 'src/common/components/OverviewCardWrapperWithSkeleton';
import { SkeletonRect } from 'src/common/components/SkeletonLoading';
import { TaskWithProgress, useTaskComplianceData } from 'src/modules/overview/taskCompliance.slice';
import { useSelectedStudyId } from 'src/modules/studies/studies.slice';
import { px, typography, colors, theme } from 'src/styles';
import { SpecColorType } from 'src/styles/theme';

const taskMap: Array<{ type: TaskType; color: SpecColorType }> = [
  { type: 'survey', color: 'secondaryTangerine' },
  { type: 'activity', color: 'secondaryViolet' },
];

const TaskComplianceCard = styled(Card)`
  height: ${px(528)};
  min-height: ${px(528)};
  color: ${colors.textPrimary} !important;
  & ${Title} {
    height: ${px(24)};
    margin-bottom: ${px(45)};
  }
`;

const SortButton = styled.button`
  width: fit-content;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  &:disabled {
    cursor: default;
    svg {
      path {
        fill: ${colors.disabled};
      }
    }
  }
`;

const BodyContainer = withCustomScrollBar(styled.div<{ rightPadding: boolean }>`
  width: 100%;
  max-height: ${px(359)};
  height: ${px(359)};
  overflow-y: auto;
  display: flex;
  flex-direction: column;
  row-gap: ${px(8)};
  padding-right: ${({ rightPadding }) => rightPadding && px(12)};
`)``;

const Footer = styled.div`
  width: 100%;
  height: ${px(24)};
  display: flex;
  justify-content: flex-start;
  align-items: center;
  column-gap: ${px(18)};
  margin-top: ${px(36)};
`;

const LegendItem = styled.div`
  ${typography.bodySmallRegular};
  width: fit-content;
  height: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding: ${px(2)};
  line-height: ${px(21)};
`;

const Circle = styled.div<{ $color: SpecColorType }>`
  width: ${px(20)};
  height: ${px(20)};
  background-color: ${({ $color }) => colors[$color]};
  border-radius: 50%;
  margin-right: ${px(10)};
`;

const EmptyScreen = styled.div`
  max-height: ${px(359)};
  height: ${px(359)};
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${colors.background};
  ${typography.headingXMediumRegular};
`;

const Legend = ({ color, type }: { color: SpecColorType; type: string }) => (
  <LegendItem key={type}>
    <Circle $color={color} />
    {type[0].toUpperCase() + type.slice(1)} tasks
  </LegendItem>
);

const Empty = () => <EmptyScreen>No data yet</EmptyScreen>;

const TaskCompliance: React.FC = () => {
  const studyId = useSelectedStudyId();
  const { data, isLoading, error, refetch } = useTaskComplianceData({
    fetchArgs: !!studyId && { studyId },
    refetchSilentlyOnMount: true,
  });

  const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>('asc');
  const [sortedData, setSortedData] = useState<TaskWithProgress[]>([]);

  const ref = useRef<HTMLDivElement>(null);
  const isContentScrolling = useScrolling(ref);

  useEffect(() => {
    if (data && data?.length) {
      setSortedData(
        [...data]?.sort((t1, t2) =>
          sortDirection === 'asc' ? t2.progress - t1.progress : t1.progress - t2.progress
        )
      );
    } else {
      setSortedData([]);
    }
  }, [data, sortDirection]);

  const extra = useMemo(
    () => (
      <SortButton
        data-testid="sort-button"
        disabled={
          !sortedData ||
          (sortedData && sortedData?.length <= 1) ||
          (!isLoading && (!data || !data?.length))
        }
        onClick={() => setSortDirection(sortDirection === 'asc' ? 'desc' : 'asc')}
      >
        {sortDirection === 'asc' ? <SortAscIcon /> : <SortDescIcon />}
        <RankIcon />
      </SortButton>
    ),
    [sortedData, setSortDirection, sortDirection, data, isLoading]
  );

  const content = useMemo(
    () =>
      sortedData?.map((t) => (
        <ProgressBar
          key={[t.id, t.type].join('-')}
          responded={t.responded || 0}
          name={t.title || ''}
          description={t.description}
          total={t.total || 1}
          percent={_round(t.progress * 100, 1)}
          color={taskMap.find(({ type }) => type === t.type)?.color || 'primaryLight'}
          isContentScrolling={isContentScrolling}
          tooltipContent={['Number of respondents', 'Completion percentage']}
          marginBottom={8}
        />
      )),
    [sortedData, isContentScrolling]
  );

  return (
    <OverviewCardWrapperWithSkeleton
      isLoading={isLoading}
      skeletons={[
        <SkeletonRect key="top" x="0" y="0" rx="2" width="63.5%" height="24" />,
        <SkeletonRect key="bottom" x="0" y="451" rx="2" width="100%" height="24" />,
      ]}
    >
      <TaskComplianceCard title="Task compliance" onReload={refetch} error={!!error} action={extra}>
        {!isLoading && (!data || !data?.length) ? (
          <Empty />
        ) : (
          <BodyContainer
            ref={ref}
            rightPadding={(sortedData?.length || 0) > 5}
            scrollbarTrackColor={theme.colors.surface}
          >
            {content}
          </BodyContainer>
        )}
        <Footer>
          {taskMap.map((t) => (
            <Legend key={t.type} color={t.color} type={t.type} />
          ))}
        </Footer>
      </TaskComplianceCard>
    </OverviewCardWrapperWithSkeleton>
  );
};

export default TaskCompliance;
