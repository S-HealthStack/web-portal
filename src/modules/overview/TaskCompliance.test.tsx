import { screen, render, act } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import userEvent from '@testing-library/user-event';
import { theme } from 'src/styles';
import { createTestStore } from '../store/testing';
import TaskCompliance from './TaskCompliance';

describe('TaskCompliance', () => {
  it('should render', async () => {
    const store = createTestStore({
      'overview/taskCompliance': {
        isLoading: false,
        fetchArgs: null,
        prevFetchArgs: null,
        data: [
          {
            id: '1',
            type: 'activity',
            revisionId: 0,
            title: 't1',
            description: 'd1',
            total: 100,
            responded: 20,
            progress: 0.2,
          },
          {
            id: '2',
            type: 'survey',
            revisionId: 0,
            title: 't2',
            description: 'd2',
            total: 100,
            responded: 20,
            progress: 0.2,
          },
        ],
      },
    });

    const result = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <TaskCompliance />
        </Provider>
      </ThemeProvider>
    );

    expect(result.baseElement).toMatchSnapshot();
    await act(() => userEvent.click(screen.getByTestId('sort-button')));
  });

  it('[NEGATIVE] should render with empty store', async () => {
    const store = createTestStore({});

    render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <TaskCompliance />
        </Provider>
      </ThemeProvider>
    );

    expect(screen.getByText('Task compliance')).toBeInTheDocument();
  });
});
