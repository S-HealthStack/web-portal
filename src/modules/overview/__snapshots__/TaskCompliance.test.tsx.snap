// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCompliance should render 1`] = `
.c18 {
  display: inline-block;
  position: relative;
}

.c17 {
  width: 4.5rem;
  min-width: 4.5rem;
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: right;
}

.c19 {
  width: 3.25rem;
  min-width: 3.25rem;
}

.c16 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 0.75rem;
  column-gap: 0.75rem;
}

.c15 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 100%;
}

.c20 {
  font-family: Inter,sans-serif;
  font-size: 0.75rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
}

.c14 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  -webkit-column-gap: 0.75rem;
  column-gap: 0.75rem;
  width: 100%;
  max-height: 1.3125rem;
  height: 1.3125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 {
  width: 100%;
  max-width: 100%;
  max-height: 2.4375rem;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-transform: capitalize;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 0.5rem;
  column-gap: 0.5rem;
  max-width: 100%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  overflow-x: hidden;
}

.c12 {
  width: 100%;
  max-width: 100%;
  max-height: 3.6875rem;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 1rem;
  margin-bottom: 0.5rem;
}

.c21 {
  width: 100%;
  height: 0.25rem;
  border-radius: 3.125rem;
  background-color: #ECF1FC;
  position: relative;
}

.c22 {
  height: 0.25rem;
  right: 0;
  border-radius: 3.125rem;
  background-color: #944ED7;
  position: relative;
}

.c23 {
  height: 0.25rem;
  right: 0;
  border-radius: 3.125rem;
  background-color: #F39C18;
  position: relative;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
}

.c0 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c8 {
  -webkit-transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  opacity: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 2.6875rem;
}

.c5 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #474747;
  display: inline-block;
  margin: 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: visible;
  position: relative;
}

.c1 {
  height: 33rem;
  min-height: 33rem;
  color: #474747 !important;
}

.c1 .c3 {
  height: 1.5rem;
  margin-bottom: 2.8125rem;
}

.c7 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.c7:disabled {
  cursor: default;
}

.c7:disabled svg path {
  fill: #DBDBDB;
}

.c9 {
  width: 100%;
  max-height: 22.4375rem;
  height: 22.4375rem;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 0.5rem;
}

.c10 {
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
  -webkit-scrollbar-color: #4475E3 #FFFFFF;
  -moz-scrollbar-color: #4475E3 #FFFFFF;
  -ms-scrollbar-color: #4475E3 #FFFFFF;
  scrollbar-color: #4475E3 #FFFFFF;
}

.c10::-webkit-scrollbar {
  width: 0.25rem;
  height: 0.25rem;
}

.c10::-webkit-scrollbar-track {
  background: #FFFFFF;
  border-radius: 0.25rem;
  margin: 0.0625rem 0;
}

.c10::-webkit-scrollbar-thumb {
  background-color: #4475E3;
  border-radius: 0.625rem;
  border: 0rem solid #FFFFFF;
  width: 0.25rem;
  min-height: 0.25rem;
}

.c24 {
  width: 100%;
  height: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 1.125rem;
  column-gap: 1.125rem;
  margin-top: 2.25rem;
}

.c25 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.125rem;
  line-height: 1.3125rem;
}

.c26 {
  width: 1.25rem;
  height: 1.25rem;
  background-color: #F39C18;
  border-radius: 50%;
  margin-right: 0.625rem;
}

.c27 {
  width: 1.25rem;
  height: 1.25rem;
  background-color: #944ED7;
  border-radius: 50%;
  margin-right: 0.625rem;
}

<body>
  <div>
    <div
      class="c0 c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3 c4"
        >
          <p
            class="c5"
            data-testid="title"
          >
            Task compliance
          </p>
          <div
            class="c6"
          >
            <button
              class="c7"
              data-testid="sort-button"
            >
              <svg />
              <svg />
            </button>
          </div>
        </div>
      </div>
      <div
        class="c8"
        data-testid="card-content"
      >
        <div
          class="c9 c10"
        >
          <div
            class="c11"
          >
            <div
              class="c12"
            >
              <div
                class="c13"
              >
                <div
                  class="c14"
                >
                  <div
                    class="c15"
                  >
                    t1
                  </div>
                  <div
                    class="c16"
                  >
                    <div
                      class="c17"
                    >
                      <span
                        class="c18"
                        data-testid="tooltip-container"
                      >
                        20
                        /
                        100
                      </span>
                    </div>
                    <div
                      class="c17 c19"
                    >
                      <span
                        class="c18"
                        data-testid="tooltip-container"
                      >
                        20
                        %
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="c15 c20"
                >
                  d1
                </div>
              </div>
              <div
                class="c21"
              >
                <div
                  class="c22"
                  style="width: 20%;"
                />
              </div>
            </div>
          </div>
          <div
            class="c11"
          >
            <div
              class="c12"
            >
              <div
                class="c13"
              >
                <div
                  class="c14"
                >
                  <div
                    class="c15"
                  >
                    t2
                  </div>
                  <div
                    class="c16"
                  >
                    <div
                      class="c17"
                    >
                      <span
                        class="c18"
                        data-testid="tooltip-container"
                      >
                        20
                        /
                        100
                      </span>
                    </div>
                    <div
                      class="c17 c19"
                    >
                      <span
                        class="c18"
                        data-testid="tooltip-container"
                      >
                        20
                        %
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="c15 c20"
                >
                  d2
                </div>
              </div>
              <div
                class="c21"
              >
                <div
                  class="c23"
                  style="width: 20%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c24"
        >
          <div
            class="c25"
          >
            <div
              class="c26"
            />
            Survey
             tasks
          </div>
          <div
            class="c25"
          >
            <div
              class="c27"
            />
            Activity
             tasks
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
